cmd/driver/main.go

```diff
@@ -110,16 +112,35 @@ func main() {
 		os.Exit(1)
 	}
 
+	metricsEndpoint := os.Getenv("METRICS_ENDPOINT")
+	if metricsEndpoint == "" {
+		level.Error(logger).Log(
+			"msg", "you need to specify an endpoint for metrics via the METRICS_ENDPOINT env var",
+		)
+		os.Exit(2)
```

Comment:
In order not to introduce a breaking change, I would either:

1. make `METRICS_ENDPOINT` optional and not expose metrics if unset, or

2. provide a default if `METRICS_ENDPOINT` is unset.

I prefer option 2 and can’t think of a reason why defaulting to `:9092` would be a bad idea.

Reply:
Only reason against using port `:9092` is the Prometheus ["Default port allocations"](https://github.com/prometheus/prometheus/wiki/Default-port-allocations) list that recommends `> 9100` for exporters.

But i suspect that this will not be an issue because anyone can just remap the port through Docker should it actually cause a collision in their setup (non k8s).

Reply:
I just reused port 9092 from your code. Feel free to change it to a recommended one.

Reply:
I reserved port `9189` for this, pushing updated code now.